tests/std/tests/Dev11_0235721_async_and_packaged_task/test.cpp

```diff
@@ -260,12 +273,45 @@ void test_shared_future_noexcept_copy_impl() {
     assert(!copyCtord.valid());
 }
 
-void test_shared_future_noexcept_copy() {
+void test_shared_future_noexcept() {
+    test_future_shared_future_noexcept_impl<int>();
+    test_future_shared_future_noexcept_impl<int&>();
+    test_future_shared_future_noexcept_impl<void>();
+
     test_shared_future_noexcept_copy_impl<int>();
     test_shared_future_noexcept_copy_impl<int&>();
     test_shared_future_noexcept_copy_impl<void>();
 }
 
+// Also test the non-constructibility of future from (future {}) and (shared_future, {})
+template <typename Void, typename T, typename... Args>
+constexpr bool is_constructible_with_trailing_empty_brace_impl = false;
+
+template <typename T, typename... Args>
+constexpr bool
+    is_constructible_with_trailing_empty_brace_impl<void_t<decltype(T(declval<Args>()..., {}))>, T, Args...> = true;
+
+template <typename T, typename... Args>
+constexpr bool is_constructible_with_trailing_empty_brace =
+    is_constructible_with_trailing_empty_brace_impl<void, T, Args...>;
+
+STATIC_ASSERT(is_constructible_with_trailing_empty_brace<pair<int, int>, int>);
+STATIC_ASSERT(is_constructible_with_trailing_empty_brace<pair<int, int>, const int&>);

```

Comment:
This is handy to prove to yourself that the trait you wrote works, but probably not necessary in the long term. (We conventionally only write tests for very complicated test machinery.)
```suggestion
```